projects/ngx-charts-on-fhir/src/lib/data-layer-list/annotation-list/annotation-list.component.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatExpansionModule } from '@angular/material/expansion';
+import { AnnotationListComponent } from './annotation-list.component';
+
+describe('DatasetAnnotationListComponent', () => {
+  let component: AnnotationListComponent;
+  let fixture: ComponentFixture<AnnotationListComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [AnnotationListComponent],
+      imports: [MatExpansionModule, MatCheckboxModule],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(AnnotationListComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  describe('onCheckboxChange', () => {
+    it('should update view, control and emit the option if an option is clicked', () => {
+      spyOn(component.annotationsChange, 'emit');
+      const annotation={ "label": { "display": true, "position": { "x": "start", "y": "end" }, "color": "#666666", "font": { "size": 16, "weight": "normal" }, "content": "Systolic Blood Pressure Reference Range" }, "type": "box", "backgroundColor": "#ECF0F9", "borderWidth": 0, "drawTime": "beforeDraw", "display": false, "yScaleID": "mm[Hg]", "yMax": 130, "yMin": 90 };
+      component.annotations = [annotation];
+      let event: any= {
+         checked : true
+      }
+      component.onCheckboxChange(annotation, event);
+      expect(component.annotationsChange.emit).toHaveBeenCalled();
```

Comment:
I see a number of issues with this test:
1. It is testing a `{ checked: true }` event for an annotation that is already set to `{ display: true }`
2. It is not checking the emitted value. I guess it's worthwhile to check that it's emitting _something_, but it would be better to check that it's emitting the _correct_ thing.
3. It is using a spy to stub the component's own `EventEmitter`. This seems like a bad idea. Why not just subscribe to the real `EventEmitter`?
4. The annotation has a lot of irrelevant properties. Please only include what is needed to make the test pass.
5. It is ignoring the component's template and testing it like a regular class. It's ok to do this sometimes, but it looks like all of your tests follow this pattern. I think we should write some tests that interact with the DOM so we'll know that everything in the template is wired up correctly. Angular Material provides test harnesses to make this easy. 

Here is how I would rewrite this test to solve all of these issues:
```ts
    it('should set display=true when checked', async () => {
      const original = [{ label: { content: 'Test' }, display: false }];
      const expected = [{ label: { content: 'Test' }, display: true }];
      let emitted: any = null;
      component.annotations = original;
      component.annotationsChange.subscribe((e) => (emitted = e));
      const checkbox = await loader.getHarness(MatCheckboxHarness);
      await checkbox.check();
      expect(emitted).toEqual(expected);
    });
```

Reply:
Hi @colin we have tried this when I am running the test case we are getting emitted as a null and the assertion is failing because expect(emitted).toEqual(expected) run before subscribe. it's not waiting to subscribe to the event.
I have coverted like this and it is working

Reply:
In your solution, the checkbox will not get checked until after the annotation changes. What will cause the annotation to change? I suspect your test is passing because `annotationsChange` never emits anything and the test completes without making any assertions.

I don't understand why my test is not working for you. The Angular Material docs say:
> The harness also automatically waits for the fixture to be stable, which will cause the test to wait for setTimeout, Promise, etc.

So it should not get past the `await checkbox.check()` until everything is resolved.

Reply:
We should probably enable jasmine's `failSpecWithNoExpectations` option so tests cannot pass without making an assertion.